sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -13,257 +13,288 @@
+        Amazon ECR in the same region specified in the `region` argument.
+    role: str
+        The ARN string of an AWS account ARN that has SageMaker execution and model registry full
+        access role.
+    instance_type: str
+        The string type of a SageMaker instance type. The default value is \"ml.c6i.xlarge\".
+    instance_count: int
+        The number of SageMaker instances to be deployed for the endpoint.
+    custom_production_variant: dict
+        The dictionary that inludes custom configuration of the SageMaker ProductionVarient
+        for identifying a model to host and the resources chosen to deploy for hosting it, as
+        documented at
+        https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ProductionVariant.html.
+    async_execution: str
+        The string boolean value, determining if using asynchronous execution mode to creating
+        endpoint. Options include "True" and "true" for True, "False" and "false" for False.
+        Default is True.
+    restore_model_path: str
+        The local folder path where GraphStorm trained model artifacts were stored, e.g.,
+        \"save_model/epoch-1/\".
+    model_yaml_config_file: str
+        The YAML file path. This YAML file is the one that was stored by GraphStorm model training
+        pipeline during model training. It is NOT the one used as one of the arguments
+        for GraphStorm model training.
+    graph_json_config_file: str
+        Path to modified GConstruct/GSProcessing JSON config. This should be available under the 
+        **output** of GConstruct/GSProcessing.
+    upload_tarfile_s3: str
+        The S3 location to upload the packed and compressed model tar file. This location should
+        be in the same region specified in the `region` argument.
+    infer_task_type: str
+        The name of real time inference task. Options include \"node_classification\".
+    model_name: str
+        A string to define the name of a SageMaker inference model. This name string must follow
+        SageMaker's naming format, which is ^[a-zA-Z0-9]([\-a-zA-Z0-9]*[a-zA-Z0-9])$. The default
+        value is \"GSF-Model4Realtime\".
+    
     """
     # ================= prepare model artifacts ================= #
     # prepare sessions
-    b3_session = boto3.Session(region_name=input_args.region)
-    sm_session = sm.Session(boto_session=b3_session)
-    
-    # If users provide the S3 tar file path already, directly use it for model creation
-    if input_args.model_tarfile_s3:
-        entrypoint_file_name = input_args.entrypoint_file_name
+    boto_session = boto3.Session(region_name=input_args.region)
+    sm_session = sm.Session(boto_session=boto_session)
 
-        model_url_s3 = input_args.model_tarfile_s3
-        assert check_tarfile_s3_object(args.model_tarfile_s3) == True, \
-            (f'Not find a tar file in the given S3 URL: {args.model_tarfile_s3}...')
+    # prepare task dependent variables
```

Comment:
Please elaborate more about with will be under current_folder.

Reply:
The support of custom mode tar file has been moved to the `tools/` folder as a courtesy tool. So will not do the check anymore. It will be users own responsibility to do sanity check.